src/gc/gcinterface.ee.h

```diff
@@ -128,6 +128,19 @@ class IGCToCLR {
     // barrier if it needs to be updated.
     virtual
     void StompWriteBarrier(WriteBarrierParameters* args) = 0;
+
+    // Signals to the finalizer thread that there are objects ready to
+    // be finalized.
+    virtual
+    void EnableFinalization() = 0;
+
+    // Returns true if there is additional work that the finalizer thread needs
+    // to do, outside of its duties of finalization.
+    virtual
+    bool HaveExtraWorkForFinalizer() = 0;
+
+    virtual
```

Comment:
This flag says whether the GC should be reporting all objects with finalizers; not just the collected ones. The `g_fFinalizeRunOnShutDown` check is somewhat warm paths - `CFinalize::RegisterForFinalization`, `CFinalize::GetNextFinalizableObject`. I do not think it should be callback to VM. Instead, the flag should stay on the GC side so that the check continues to be fast and there should be a method on JIT/EE interface to set it. 

Note that you should not ever see this flag set in CoreCLR. It is only used in full Framework.